modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java

```diff
@@ -0,0 +1,379 @@
+import com.liferay.portal.kernel.service.LayoutLocalServiceUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.test.util.UserTestUtil;
+import com.liferay.portal.kernel.util.LinkedHashMapBuilder;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.segments.service.SegmentsExperienceLocalService;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.Serializable;
+
+import java.util.Map;
+import java.util.Scanner;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Daniel Szimko
+ */
+@RunWith(Arquillian.class)
+public class LayoutFromMasterPageTemplateExportImportTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	public void importLayouts(
+			Map<String, String[]> parameterMap, boolean expectError)
+		throws Exception {
+
+		try (LogCapture logCapture = getLogCapture(expectError)) {
+			User user = TestPropsValues.getUser();
+
+			Map<String, Serializable> importLayoutSettingsMap =
+				ExportImportConfigurationSettingsMapFactoryUtil.
+					buildImportLayoutSettingsMap(
+						user, _importedGroup.getGroupId(), false, null,
+						parameterMap);
+
+			ExportImportConfiguration exportImportConfiguration =
+				ExportImportConfigurationLocalServiceUtil.
+					addExportImportConfiguration(
+						user.getUserId(), _importedGroup.getGroupId(),
+						StringPool.BLANK, StringPool.BLANK,
+						ExportImportConfigurationConstants.TYPE_IMPORT_LAYOUT,
+						importLayoutSettingsMap, WorkflowConstants.STATUS_DRAFT,
+						new ServiceContext());
+
+			ExportImportServiceUtil.importLayouts(
+				exportImportConfiguration, larFile);
+		}
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+		_importedGroup = GroupTestUtil.addGroup();
+
+		UserTestUtil.setUser(TestPropsValues.getUser());
+	}
+
+	@Test
+	public void testExportImportLayoutFromMasterTemplateAndDraftLayoutMappingOnImportSide()
+		throws Exception {
+
+		// make sure that the layout ID is the same as layout4's layoutId
+
+		LayoutTestUtil.addTypePortletLayout(_group, true);
```

Comment:
Is it necessary to create 5 to reproduce the issue? Can't we just simplify and create the minimum possible to reproduce it?